tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_gemv_kk.hpp

```diff
@@ -0,0 +1,231 @@
+
+#ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_GEMV_HPP_
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_GEMV_HPP_
+
+#include "signal_kokkos_impl_called.hpp"
+
+namespace KokkosKernelsSTD {
+
+namespace gemv_impl{
+template<class Accessor>
+double get_scaling_factor(Accessor) { return 1.0; }
```

Comment:
This will force the scaling factor to have type `double`, even if no floating-point values are involved.  (What if all the matrices have element type `int64_t`, for example?)

Would you consider having `get_scaling_factor` take both an accessor type and a value type?  The value type could come from the calling context.  For example, in `y = A * x`, the scaling factor could have type `declval<typename decltype(A)::value_type>() * declval<typename decltype(x)::value_type>()`.

Reply:
@mhoemmen this code is here because it was added by Christian, so I did not touch it :) 
I can make the changes you suggest but just wanted to point out I have not touched this code for that reason!

Reply:
in fact, I also left the original specialization, see last overload in that header:
```
//
// the following specialization is here to avoid breaking
// the example code started by CTrott in this file:
//    stdBLAS/examples/kokkos-based/matrix_vector_product_kokkos.cpp
// but we need to change this later after we properly handle accessors
```

Reply:
@mhoemmen i made some changes